packages/components/src/presentation-components/tree/controlled/UsePresentationTreeState.ts

```diff
@@ -213,6 +220,11 @@ function useTreeState(props: UseTreeStateProps) {
     onNodeLoadedRef.current = props.onNodeLoaded;
   }, [props.onNodeLoaded]);
 
+  const onHierarchyLimitExceededRef = useRef(props.onHierarchyLimitExceeded);
+  useEffect(() => {
+    onHierarchyLimitExceededRef.current = props.onHierarchyLimitExceeded;
+  }, [props.onHierarchyLimitExceeded]);
```

Comment:
This pattern is repeated 3 times here. Can it be generalized and extracted into custom hook?

Reply:
Extracted into a `useLatest` hook.